frontend/packages/dev-console/src/components/import/pipeline/PipelineTemplate.tsx

```diff
@@ -0,0 +1,69 @@
+import * as React from 'react';
+import { ExpandCollapse, LoadingInline } from '@console/internal/components/utils';
+import { useFormikContext, FormikValues } from 'formik';
+import { Alert } from '@patternfly/react-core';
+import { PipelineVisualization } from '../../pipelines/PipelineVisualization';
+import { getPipelineTemplate, getPipelineTaskRefs } from './pipeline-template-utils';
+
+const PipelineTemplate: React.FC = () => {
+  const [noTemplateForRuntime, setNoTemplateForRuntime] = React.useState(false);
+  const {
+    values: {
+      pipeline: { template },
+      image,
+    },
+    setFieldValue,
+  } = useFormikContext<FormikValues>();
+
+  React.useEffect(() => {
+    let ignore = false;
+
+    const fetchPipelineTemplate = async () => {
+      const pipelineTemplate = await getPipelineTemplate(image.selected);
+      if (!ignore && pipelineTemplate) {
+        const taskRefs = getPipelineTaskRefs(pipelineTemplate);
+        setFieldValue('pipeline.template', pipelineTemplate);
+        setFieldValue('pipeline.taskRefs', taskRefs);
+        setNoTemplateForRuntime(false);
+      } else {
+        setFieldValue('pipeline.template', null);
+        setFieldValue('pipeline.taskRefs', null);
+        setNoTemplateForRuntime(true);
+      }
+    };
+
+    fetchPipelineTemplate();
+
+    return () => {
+      ignore = true;
+    };
```

Comment:
It will only ignore if `image.selected` or `setFieldValue` changes.
I think this should also check for when the component is unmounted while the async is happening otherwise it'll still try to call `setFieldValue` and if that doesn't protected against unmounting there'll be errors.

Reply:
React calls the function that is returned by `useEffect` hook when the component is unmounted. So, that's when value of `ignore` is changed to `true`. If it's true while the async call is still to be returned, no state is updated.

Reply:
well _now_ it works this way :P